extensions/amp-image-slider/1.0/component.js

```diff
@@ -0,0 +1,701 @@
+/**
+ * Copyright 2021 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {createElementWithAttributes} from '#core/dom';
+import {htmlFor} from '#core/dom/static-template';
+import {setStyle} from '#core/dom/style';
+import {clamp} from '#core/math';
+
+import * as Preact from '#preact';
+import {
+  useCallback,
+  useEffect,
+  useLayoutEffect,
+  useMemo,
+  useRef,
+  useState,
+} from '#preact';
+import {forwardRef} from '#preact/compat';
+
+// import {Services} from '#service';
+
+import {useStyles} from './component.jss';
+
+import {listen} from '../../../src/event-helper';
+import {Gestures} from '../../../src/gesture';
+import {SwipeXRecognizer} from '../../../src/gesture-recognizers';
+
+const VALID_IMAGE_TAGNAMES = new Set(['AMP-IMG', 'IMG']);
```

Comment:
This is too prescriptive for the Preact version -- not only is `amp-img` specific to AMP, but users of this component may want to provide their own image-based components to compose with.

Reply:
Got it. I will remove this check for Preact version :)